vars/commonPipelineEnvironment.groovy

```diff
@@ -1,51 +1,41 @@
 import com.sap.piper.ConfigurationLoader
 import com.sap.piper.ConfigurationMerger
+import com.sap.piper.DefaultValueCache
 import com.sap.piper.analytics.InfluxData
 
 class commonPipelineEnvironment implements Serializable {
 
-    //stores version of the artifact which is build during pipeline run
-    def artifactVersion
-
-    //Stores the current buildResult
-    String buildResult = 'SUCCESS'
-
-    //stores the gitCommitId as well as additional git information for the build during pipeline run
-    String gitCommitId
-    String gitCommitMessage
-    String gitSshUrl
-    String gitHttpsUrl
-    String gitBranch
-
-    //GiutHub specific information
-    String githubOrg
-    String githubRepo
-
-    //stores properties for a pipeline which build an artifact and then bundles it into a container
-    private Map appContainerProperties = [:]
-
-    Map configuration = [:]
-    Map defaultConfiguration = [:]
+    //
+    // We forward to cpe declared on DefaultValueCache
+    def methodMissing(String name, def args) {
+        DefaultValueCache.commonPipelineEnvironment.invokeMethod(name, args)
+    }
 
-    String mtarFilePath
-    private Map valueMap = [:]
+    def propertyMissing(def name) {
+       DefaultValueCache.commonPipelineEnvironment[name]
+    }
 
-    void setValue(String property, value) {
-        valueMap[property] = value
+    def propertyMissing(def name, def value) {
+       DefaultValueCache.commonPipelineEnvironment[name] = value
     }
+    // End forwarding to DefaultValueCache
+    //
 
-    def getValue(String property) {
-        return valueMap.get(property)
+    /*
+     * Should only be used by tests
+     */
+    void setConfiguration(Map configuration) {
```

Comment:
likley does not cater for secnarios like

` commonPipelineEnvironment.configuration.general.whatever = 'something'`